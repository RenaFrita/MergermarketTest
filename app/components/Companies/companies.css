.companies {
  border: 2px solid #000;
  list-style-type: none;
  padding: 0;
}

.companies li {
  border-bottom: 2px solid #000;
}

.companies li:last-child {
  border-bottom: 0;
}

.companies-headers {
  background-color: #d3d3d3;
  display: flex;
  vertical-align: middle;
}

.companies-headers span {
  border-left: 2px solid #000;
  box-sizing: border-box;
  display: inline-block;
  font-weight: 700;
  padding: 10px;
  flex: 1;
  word-break: break-word;
}

.companies-headers span:first-child {
  border-left: 0;
  flex: 6;
}
