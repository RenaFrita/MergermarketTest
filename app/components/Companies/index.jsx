import React from 'react';
import PropTypes from 'prop-types';
import Company from '../Company/index.jsx';
import './companies.css';

export function type(listed, pevc){
  return listed ? pevc ?
    <div><span>Listed Entity</span><span>PEVC House</span></div> :
    <span>Listed Entity</span> : '';
}

export function headquarter(geography) {
  return `${geography.state ? geography.state : ''}${geography.state && geography.country ? ', ': ''}${geography.country ? geography.country: ''}`;
}

const Companies = ({ companies, show }) => {
  const showMore = show ?
    <li>
      <a className="companies-more" onClick={show}>
        Show more
      </a>
    </li> : '';

  return (
    <ul className="companies">
      <li className="companies-headers">
        <span>Name</span>
        <span>Type</span>
        <span>Headquarter</span>
      </li>
      {
        companies.map((company) => {
          return (
            <li key={company.id}>
              <Company
                name={company.name}
                description={company.description}
                type={type(company.isListedEntity, company.isPEVCHouse)}
                headquarter={headquarter(company.geography)}
              />
            </li>
          );
        })
      }
      {showMore}
    </ul>
  );
}

Companies.propTypes = {
  companies: PropTypes.array.isRequired,
  show: PropTypes.func,
};

export default Companies;
