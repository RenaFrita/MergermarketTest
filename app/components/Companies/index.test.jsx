import React from 'react';
import Companies, { type, headquarter } from './index.jsx';
import renderer from 'react-test-renderer';
import companies from '../../../data/company';

describe('Company component', () => {
  test('should render the correct markup', () => {
    const mockFn = jest.fn();
    const component = renderer.create(
      <Companies
        companies={companies}
        show={mockFn}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('should return empty string', () => {
    let result = type(false, false);
    expect(result).toBe('');
  })

  test('should return a node with ', () => {
    let result = type(true, false);
    expect(result).toMatchObject(<span>Listed Entity</span>);
  })

  test('should return a node with ', () => {
    let result = type(true, true);
    expect(result).toMatchObject(<div><span>Listed Entity</span><span>PEVC House</span></div>);
  })

  test('should return a string with the state and country', () => {
    expect(headquarter({ state: 'Kanto', country: 'Japan'})).toBe('Kanto, Japan');
  });
});
