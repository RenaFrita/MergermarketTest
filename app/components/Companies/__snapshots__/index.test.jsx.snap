// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company component should render the correct markup 1`] = `
<ul
  className="companies"
>
  <li
    className="companies-headers"
  >
    <span>
      Name
    </span>
    <span>
      Type
    </span>
    <span>
      Headquarter
    </span>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Corporation
        </h2>
        <p
          className="company-description"
        >
          Japan-based company engaged in businesses of consumer and professional electronics, gaming, entertainment and financial services
        </p>
      </header>
      <p
        className="company-type"
      >
        <span>
          Listed Entity
        </span>
      </p>
      <p
        className="company-headquarter"
      >
        Kanto, Japan
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Music Entertainment Inc.
        </h2>
        <p
          className="company-description"
        >
          US-based music publishing company
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        New York (NY), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Mobile Communications AB
        </h2>
        <p
          className="company-description"
        >
          UK-based company engaged in product research, development, design, production, distribution, marketing, sale and customer servicing of mobile phones, accessories and applications
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        Greater London, United Kingdom
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Pictures Entertainment Inc.
        </h2>
        <p
          className="company-description"
        >
          US-based media company involved in motion picture production and distribution, television programming and syndication, home video acquisition and distribution, operation of studio facilities, development of new entertainment technologies and distribution of filmed entertainment
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        California (CA), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Network Communications Inc. 
        </h2>
        <p
          className="company-description"
        >
          Japan based internet service provider
        </p>
      </header>
      <p
        className="company-type"
      >
        <span>
          Listed Entity
        </span>
      </p>
      <p
        className="company-headquarter"
      >
        Kanto, Japan
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Pictures Networks India Pvt. Ltd.
        </h2>
        <p
          className="company-description"
        >
          India-based media company
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        Maharashtra, India
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony/ATV Music Publishing LLC
        </h2>
        <p
          className="company-description"
        >
          US-based music publisher
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        New York (NY), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Financial Holdings Inc.
        </h2>
        <p
          className="company-description"
        >
          Japan based arm of Sony Group Financial controling life insurance companies, non-life insurance companies and banks
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        Kanto, Japan
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Pictures Television International
        </h2>
        <p
          className="company-description"
        >
          US-based producer and distributor of films and television networks
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        California (CA), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Sony Ericsson Mobile Communications AB
        </h2>
        <p
          className="company-description"
        >
          US based automotive and industrial wireless solutions provider
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        North Carolina (NC), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          IBM Corporation
        </h2>
        <p
          className="company-description"
        >
          US-based provider of IT products and services
        </p>
      </header>
      <p
        className="company-type"
      >
        <span>
          Listed Entity
        </span>
      </p>
      <p
        className="company-headquarter"
      >
        New York (NY), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          IBM Global Services
        </h2>
        <p
          className="company-description"
        >
          IBM IT consultancy 
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          Grupo Ibmec Educacional S.A.
        </h2>
        <p
          className="company-description"
        >
          Brazil-based educational institution that provides graduation programs including certification program in business administration
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        Sao Paulo, Brazil
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          IBM (x86 server business)
        </h2>
        <p
          className="company-description"
        >
          US-based unit comprises IBM's System x line, which includes System x, BladeCenter and Flex System blade servers and switches, x86-based Flex integrated systems, NeXtScale and iDataPlex servers and associated software, blade networking and maintenance operations
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        New York (NY), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          IBM Credit LLC
        </h2>
        <p
          className="company-description"
        >
          US-based leasing company
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        New York (NY), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          HGST, Inc.
        </h2>
        <p
          className="company-description"
        >
          US-based manufacturer and developer of hard disk drives and components for PCs, servers, and electronic devices
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        California (CA), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          IBM Corporation (Personal Computing Division)
        </h2>
        <p
          className="company-description"
        >
          US based personal computing division of IBM Corporation.
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          IBM Corporation (semiconductor manufacturing operations)
        </h2>
        <p
          className="company-description"
        >
          US-based chip manufacturing and semiconductor unit of IBM Corporation
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        New York (NY), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          The IBM Watson Group
        </h2>
        <p
          className="company-description"
        >
          US-based investment holding company having interest in operators of intelligence technology platform for creating interactive virtual agents to chat
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        New York (NY), USA
      </p>
    </article>
  </li>
  <li>
    <article
      className="company"
    >
      <header
        className="company-details"
      >
        <h2
          className="company-name"
        >
          IBM Japan Services Company Ltd.
        </h2>
        <p
          className="company-description"
        >
          Japan-based company engaged in the provision of computer hardware and software products, consulting, systems integration and other technology services
        </p>
      </header>
      <p
        className="company-type"
      >
        
      </p>
      <p
        className="company-headquarter"
      >
        Kanto, Japan
      </p>
    </article>
  </li>
  <li>
    <a
      className="companies-more"
      onClick={[Function]}
    >
      Show more
    </a>
  </li>
</ul>
`;
